### PR TITLE
Migration Doc (v8 to v10)

### PR DESCRIPTION
**Pull request by rishabpoh**
Publishing the existing migration doc to master branch.

### CODE REVIEW

File: MigrationGuideV8toV10.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Check the MD5 hash when downloading data | `/CheckMD5` | `--check-md5=[option]` <br> Options: NoCheck, LogOnly, FailIfDifferent (default), FailIfDifferentOrMissing
+Retrieve listing | `/L` | `azcopy list`
+Set modified time to be same as the source blobs | `/MT` | `--preserve-last-modified-time`
+Exclude newer source | `/XN` | Not yet supported
+Exclude older source | `/XO` | Use the sync command
+Upload archive files/blobs | `/A` | See row below
+Set attributes | `/IA:[RASHCNETOI]` <br> `/XA:[RASHCNETOI]` | `--include-attributes string` <br> `--exclude-attributes string`
+Copy blobs or files synchronously among two Azure Storage endpoints | `/SyncCopy` | V10 is always synchronous from source to destination, unlike v8 which downloads then re-uploads
+Set content type | `/SetContentType:[content-type]` | `--content-type string`
+Set blob type at destination | `/BlobType:<option>` Options: page, block, append | `--blob-type string`
```
Comment (JohnRusk) [inline MigrationGuideV8toV10.md:121]: Maybe note what the valid values are for v10.  IIRC they are not identical to the v8 ones.

File: MigrationGuideV8toV10.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+### How can I use AzCopy to work with Azure Tables?
+The latest version that supports tables is AzCopy 7.3.
+
+### How can I use AzCopy to work with Azure Queues?
+The latest version that supports queues is AzCopy 8.1.
+
+### What is different between the job management (how to resume jobs in both use examples)?
+In v8, you can rerun the same command from before and answer the prompt.
+In v10, you can have a job sub group where you can resume with job id.
```
Comment (JohnRusk) [inline MigrationGuideV8toV10.md:140]: Maybe link to: https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-jobs-resume

**Comment by zezha-msft** [comment]
@aishmanoh please take a look when you have time. This is a great doc to add into the repo but some comments need to be addressed.

File: MigrationGuideV8toV10.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+Match a specific pattern | `/Pattern:<pattern>` | `--include-pattern string` <br> `--exclude-pattern string` <br> `--include-path string` <br> `--exclude-path string`
+Create an MD5 hash when downloading data | Always does this | `--put-md5`
+Check the MD5 hash when downloading data | `/CheckMD5` | `--check-md5=[option]` <br> Options: NoCheck, LogOnly, FailIfDifferent (default), FailIfDifferentOrMissing
+Retrieve listing | `/L` | `azcopy list`
+Set modified time to be same as the source blobs | `/MT` | `--preserve-last-modified-time`
+Exclude newer source | `/XN` | Not yet supported
+Exclude older source | `/XO` | Use the sync command
+Upload archive files/blobs | `/A` | See row below
+Set attributes | `/IA:[RASHCNETOI]` <br> `/XA:[RASHCNETOI]` | `--include-attributes string` <br> `--exclude-attributes string`
+Copy blobs or files synchronously among two Azure Storage endpoints | `/SyncCopy` | V10 is always synchronous from source to destination, unlike v8 which downloads then re-uploads
```
[the remark this answers is not in the document]
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:119]: Done

File: MigrationGuideV8toV10.md (hunk 2 above)
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:140]: Done

File: MigrationGuideV8toV10.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+Authenticate | `/SourceKey:<Key>` <br> `/DestKey:<Key>` <br> `/SourceSAS:<SAS>` <br> `/DestSAS:<SAS>` | Append the SAS token to the source and/or destination URI. <br> `'blob URI' + '?' + 'SAS'`
+Check log verbosity | `/V:[verbose-log-file]` | `--log-level`
+Specify journal file folder | `/Z:[journal-file-folder]` | Modify environment variable: <br> `AZCOPY_JOB_PLAN_LOCATION`
+Specify parameter file | `/@:<parameter-file>` | Run commands in command line
+Suppress confirmation prompts | `/Y` | Suppressed by default. To enable, specify parameter: INSERT HERE
+Specify number of concurrent operations | `/NC:<number-of-concurrent>` | Modify environment variable: <br> `AZCOPY_CONCURRENCY_VALUE`
+Specify source/destination type | `/SourceType:<option>` `/DestType:<option>` Options: blob, file | `--from-to=[enums]` <br> (typically not used)
+Upload contents recursively | `/S` | `--recursive`
+Match a specific pattern | `/Pattern:<pattern>` | `--include-pattern string` <br> `--exclude-pattern string` <br> `--include-path string` <br> `--exclude-path string`
+Create an MD5 hash when downloading data | Always does this | `--put-md5`
```
[the remark this answers is not in the document]
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:111]: Done

File: MigrationGuideV8toV10.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+Check the MD5 hash when downloading data | `/CheckMD5` | `--check-md5=[option]` <br> Options: NoCheck, LogOnly, FailIfDifferent (default), FailIfDifferentOrMissing
```
[the remark this answers is not in the document]
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:112]: Done

File: MigrationGuideV8toV10.md (hunk 1 above)
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:121]: Done

File: MigrationGuideV8toV10.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  --recursive
+```
+
+## Parameters Table
+
+Task | v8 | v10
+------------ | ------------- | -------------
+Authenticate | `/SourceKey:<Key>` <br> `/DestKey:<Key>` <br> `/SourceSAS:<SAS>` <br> `/DestSAS:<SAS>` | Append the SAS token to the source and/or destination URI. <br> `'blob URI' + '?' + 'SAS'`
+Check log verbosity | `/V:[verbose-log-file]` | `--log-level`
+Specify journal file folder | `/Z:[journal-file-folder]` | Modify environment variable: <br> `AZCOPY_JOB_PLAN_LOCATION`
```
[the remark this answers is not in the document]
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:104]: Done

File: MigrationGuideV8toV10.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+Specify parameter file | `/@:<parameter-file>` | Run commands in command line
+Suppress confirmation prompts | `/Y` | Suppressed by default. To enable, specify parameter: INSERT HERE
```
[the remark this answers is not in the document]
Comment (aishmanoh) [reply MigrationGuideV8toV10.md:106]: Done

**Review by zezha-msft — APPROVED** [review]

**Comment by mohsha-msft** [comment]
@zezha-msft 
FYI: Merged directly to `master`?

**Comment by zezha-msft** [comment]
Yes @mohsha-msft, since it's a doc independent of 10.10.

**Pull request merged**